llms/googleai/embeddings.go

```diff
@@ -11,12 +12,20 @@ func (g *GoogleAI) CreateEmbedding(ctx context.Context, texts []string) ([][]flo
 	em := g.client.EmbeddingModel(g.opts.DefaultEmbeddingModel)
 
 	results := make([][]float32, 0, len(texts))
-	for _, t := range texts {
-		res, err := em.EmbedContent(ctx, genai.Text(t))
-		if err != nil {
-			return results, err
+
+	batch := em.NewBatch()
+	for i, t := range texts {
+		batch = batch.AddContent(genai.Text(t))
+		if (i > 0 && (i+1)%100 == 0) || i == len(texts)-1 {
+			embeddings, err := em.BatchEmbedContents(ctx, batch)
+			if err != nil {
+				return nil, fmt.Errorf("failed to create embeddings: %w", err)
+			}
+			for _, e := range embeddings.Embeddings {
+				results = append(results, e.Values)
+			}
+			batch = em.NewBatch()
 		}
```

Comment:
The batching API allows 100 documents per batch, this automatically sends a request every 100 texts. I can add a comment if you prefer?

Reply:
Added a comment.